private void Handle(Exception ex) { }